core/trino-main/src/test/java/io/trino/sql/query/QueryAssertions.java

```diff
@@ -866,13 +872,38 @@ VALUES ROW(%s)
                 fail("Mismatched results between interpreter and evaluation engine: %s vs %s".formatted(full.value(), withConstantFolding.value()));
             }
 
-            return new Result(full.type(), full.value);
+            return new Result(full.type(), full.value, full.expression);
+        }
+
+        private static Optional<Expression> extractExpressionWithBindings(Plan plan)
+        {
+            Optional<PlanNode> planNode = PlanNodeSearcher.searchFrom(plan.getRoot())
+                    .whereIsInstanceOfAny(ProjectNode.class)
+                    .findFirst();
+            if (planNode.isPresent() && planNode.get() instanceof ProjectNode projectNode) {
```

Comment:
use:
```
return PlanNodeSearcher.searchFrom(plan.getRoot())
  .whereIsInstanceOfAny(ProjectNode.class)
  .<ProjectNode>findFirst();
  .flatMap(node -> node.getAssignments().getExpressions().stream().findFirst())
```

Reply:
`findFirst()` does not use generics and change this is rather not in scope of this PR.